middleware/git/git.go

```diff
@@ -0,0 +1,167 @@
+package git
+
+import (
+	"fmt"
+	"github.com/mholt/caddy/middleware"
+	"log"
+	"net/http"
+	"net/url"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Git represents a middleware instance that pulls git repository.
+type Git struct {
+	Next middleware.Handler
+	Repo *Repo
+}
+
+// ServeHTTP satisfies the middleware.Handler interface.
+func (g Git) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error) {
+	if err := g.Repo.Pull(); err != nil {
+		log.Println(err)
```

Comment:
Conveniently, this line is not needed. Just return the err like you do in the next line. Error handling middleware will take care of logging it.